src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.GeneratedFileReplacingCompilationTracker.cs

```diff
@@ -19,8 +18,10 @@ internal partial class SolutionCompilationState
     /// to return a generated document with a specific content, regardless of what the generator actually produces. In other words, it says
     /// "take the compilation this other thing produced, and pretend the generator gave this content, even if it wouldn't."
     /// </summary>
-    private class GeneratedFileReplacingCompilationTracker(ICompilationTracker underlyingTracker, TextDocumentStates<SourceGeneratedDocumentState> replacementDocumentStates) : ICompilationTracker
+    private class GeneratedFileReplacingCompilationTracker : ICompilationTracker
```

Comment:
had to move off of primary constructor as the non-copyable analyzer was throwing up false positives with the initialization of the SkeletonReferenceCache  field.